Userland/Libraries/LibWeb/FileAPI/Blob.cpp

```diff
@@ -238,20 +241,28 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Blob>> Blob::slice(Optional<i64> start, Opt
     // a. S refers to span consecutive bytes from this, beginning with the byte at byte-order position relativeStart.
     // b. S.size = span.
     // c. S.type = relativeContentType.
-    auto byte_buffer = TRY_OR_THROW_OOM(realm().vm(), m_byte_buffer.slice(relative_start, span));
+    auto byte_buffer = TRY_OR_THROW_OOM(vm, m_byte_buffer.slice(relative_start, span));
     return MUST_OR_THROW_OOM(heap().allocate<Blob>(realm(), realm(), move(byte_buffer), move(relative_content_type)));
 }
 
 // https://w3c.github.io/FileAPI/#dom-blob-text
 JS::Promise* Blob::text()
 {
+    auto& realm = this->realm();
+
     // FIXME: 1. Let stream be the result of calling get stream on this.
     // FIXME: 2. Let reader be the result of getting a reader from stream. If that threw an exception, return a new promise rejected with that exception.
 
     // FIXME: We still need to implement ReadableStream for this step to be fully valid.
     // 3. Let promise be the result of reading all bytes from stream with reader
-    auto promise = JS::Promise::create(realm());
-    auto result = JS::PrimitiveString::create(vm(), DeprecatedString { m_byte_buffer.bytes() });
+    auto promise = JS::Promise::create(realm);
+    auto result_or_error = JS::PrimitiveString::create(vm(), StringView { m_byte_buffer.bytes() });
+    if (result_or_error.is_error()) {
+        auto error = WebIDL::OperationError::create(realm, "Failed to read bytes.");
+        promise->reject(error);
+        return promise;
+    }
```

Comment:
This is a regular OOM error, why would we treat it specially?

Reply:
I've said this before on another PR (someone else's, probably) - I don't think rejecting promises in an OOM situation makes much sense, processing them will likely allocate as well.